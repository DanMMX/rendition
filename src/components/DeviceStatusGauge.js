import * as React from 'react'
const filter = require('lodash/filter')
import { withTheme } from 'styled-components'
import { compose } from 'recompose'
import * as ResinDeviceStatus from 'resin-device-status'
import Gauge from './Gauge'

import asRendition from '../asRendition'
import theme from '../theme'

const { colors } = theme

const pathFillColor = status => {
  if (status === 'idle') {
    return colors.statusIdle.main
  }

  if (status === 'configuring') {
    return colors.statusConfiguring.main
  }

  if (status === 'updating') {
    return colors.statusUpdating.main
  }

  if (status === 'post-provisioning') {
    return colors.statusPostProvisioning.main
  }

  if (status === 'offline') {
    return colors.statusOffline.main
  }
}

const DeviceStatusGauge = ({ devices, ...props }) => {
  const data = []
  const allDevices = devices || []

  ResinDeviceStatus.statuses.forEach(status => {
    const devices = filter(
      allDevices,
      d => ResinDeviceStatus.getStatus(d).key === status.key
    )

    data.push({
      key: status.key,
      name: status.name,
      value: devices.length,
      color: pathFillColor(status.key)
    })
  })

  return (
    <Gauge
      data={data}
      title='Total Devices'
      placeholderColor={colors.statusOffline.main}
    />
  )
}

export default compose(withTheme, asRendition)(DeviceStatusGauge)
