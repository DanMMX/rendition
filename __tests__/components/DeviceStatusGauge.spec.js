/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import DeviceStatusGauge from '../../src/components/DeviceStatusGauge'

test('DeviceStatusGauge renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <DeviceStatusGauge>DeviceStatusGauge</DeviceStatusGauge>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
