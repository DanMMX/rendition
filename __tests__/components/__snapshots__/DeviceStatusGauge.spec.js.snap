// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceStatusGauge renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <div
    className="sc-dnqmqq gDpXJY sc-gzVnrw iYSyBq"
  >
    <svg
      className="gauge__graph"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill=" white"
        r="46"
        stroke="#dadada"
        strokeWidth="5"
      />
      <g>
        <circle
          cx="50"
          cy="50"
          fill="#fd7c7c"
          r="46"
        />
        <circle
          cx="50"
          cy="50"
          fill="white"
          r="25"
        />
      </g>
      <g
        transform="translate(50, 50)"
      />
      <text
        className="gauge__graph__total-label"
        x="50"
        y="41"
      >
         TOTAL DEVICES
      </text>
      <text
        className="sc-htoDjs gNiaxF"
        x="50"
        y="63"
      >
        0
      </text>
    </svg>
    <ul
      className="gauge__legend"
    >
      <li
        className="gauge__legend-item"
      >
        <span
          className="gauge__legend-color"
          style={
            Object {
              "backgroundColor": "#89c683",
            }
          }
        >
           
        </span>
        <span
          className="gauge__legend-name"
        >
          Online
        </span>
      </li>
      <li
        className="gauge__legend-item"
      >
        <span
          className="gauge__legend-color"
          style={
            Object {
              "backgroundColor": "#ffb25e",
            }
          }
        >
           
        </span>
        <span
          className="gauge__legend-name"
        >
          Configuring
        </span>
      </li>
      <li
        className="gauge__legend-item"
      >
        <span
          className="gauge__legend-color"
          style={
            Object {
              "backgroundColor": "#7ccdfd",
            }
          }
        >
           
        </span>
        <span
          className="gauge__legend-name"
        >
          Updating
        </span>
      </li>
      <li
        className="gauge__legend-item"
      >
        <span
          className="gauge__legend-color"
          style={
            Object {
              "backgroundColor": "#fd7c7c",
            }
          }
        >
           
        </span>
        <span
          className="gauge__legend-name"
        >
          Offline
        </span>
      </li>
      <li
        className="gauge__legend-item"
      >
        <span
          className="gauge__legend-color"
          style={
            Object {
              "backgroundColor": "#aa96d5",
            }
          }
        >
           
        </span>
        <span
          className="gauge__legend-name"
        >
          Post Provisioning
        </span>
      </li>
    </ul>
  </div>
</div>
`;
